kube-client/src/client/mod.rs

```diff
@@ -243,6 +243,7 @@ impl Client {
     /// and [`AsyncBufReadExt`](futures::AsyncBufReadExt).
     pub async fn request_stream(&self, request: Request<Vec<u8>>) -> Result<impl AsyncBufRead> {
         let res = self.send(request.map(Body::from)).await?;
+        let res = handle_api_errors(res).await?;
```

Comment:
i think this refactor makes sense, but i would rather this was its own PR so that it shows up in the changelog as a FIX rather than a random addition for a new feature

Reply:
Yes I thought so, will do it on another PR